### PR TITLE
fix: Fix resource `protocols` diffing and serialization

### PR DESCRIPTION
**Pull request by ekampf**
## Related Tickets & Documents

- Issue: #145 

## Changes

Fixed `Resource.is_matching_spec` - it was not comparing `protocols` properly so would always return false causing `twingate_resource_sync` to always think there's a drift and issue an update call

### CODE REVIEW

File: app/api/client_resources.py [hunk 1]
```diff
@@ -115,25 +115,27 @@ def is_matching_spec(self, crd: ResourceSpec) -> bool:
...
-            name=self.name,
-            address=self.address.value,
-            alias=self.alias,
-            is_visible=self.is_visible,
-            is_browser_shortcut_enabled=self.is_browser_shortcut_enabled,
-            remote_network_id=self.remote_network.id,
-            security_policy_id=(
-                self.security_policy.id if self.security_policy else None
-            ),
-            protocols=self.protocols,
```
Comment (ekampf) [inline app/api/client_resources.py:136]: Bug was here - `self.protocols` is an object so we need to serialize it and not send it as is.
Doing `self.protocols.dict()` here would work but a safer approach (what happens if we have another such property in the future?) is to use `self.dict(` and give it the fields we want and it'll take care of it

File: app/crds.py [hunk 2]
```diff
@@ -123,7 +123,11 @@ class ResourceSpec(BaseModel):
     security_policy_id: str | None = None
     is_visible: bool = True
     is_browser_shortcut_enabled: bool = True
-    protocols: ResourceProtocols | None = None
+    protocols: ResourceProtocols = ResourceProtocols(
```
Comment (ekampf) [inline app/crds.py:126]: Added default to replicate API behavior - when no protocols limitation is set, reading resource from the API will return a protocols object like this (and not `null`)

File: app/api/client_resources.py [hunk 3]
```diff
@@ -235,7 +244,7 @@ def resource_create(
                 "isBrowserShortcutEnabled": resource.is_browser_shortcut_enabled,
                 "remoteNetworkId": resource.remote_network_id,
                 "securityPolicyId": resource.security_policy_id,
-                "protocols": resource.protocols,
+                "protocols": resource.protocols.dict(by_alias=True),
```
Comment (ekampf) [inline app/api/client_resources.py:247]: `resource.protocols` is a PyDantic object which is not JSON serializable.
We didnt have a test path here send sends a non-null value before so this never failed

File: pyproject.toml [hunk 4]
```diff
@@ -115,7 +115,7 @@ markers = [
     "integration",
 ]
 
-[tool.ruff]
```
Comment (ekampf) [inline pyproject.toml:118]: Fix ruff warnings re configuration location change

**Review by liorr — APPROVED** [review]
LGTM

**Pull request merged**